### ISSUE TITLE
arguments-differ message doesn't reflect recent change

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#1041]
Since 37889449a00cefa40c23b2254a3ffcdc5898c635 for something like:

``` python
class UsageFormatter(argparse.HelpFormatter):

    def _format_usage(self, usage, actions, groups, _prefix):
        """Override _format_usage to not add the 'usage:' prefix."""
        return super()._format_usage(usage, actions, groups, '')
```

I get "W: 54, 4: Arguments number differs from overridden '_format_usage' method (arguments-differ)"

I think there are two issues there:
- The message should say "Arguments differ from ..." or so?
- I'm not sure if the change actually should be there. In my case, I know it will never be called with `prefix=...` - though indeed it could cause a breakage with a Python update... On the other hand, how would I now mark the `_prefix` argument as unused, except with a disable pragma?

**Comment by The-Compiler** [comment]
Another example of a similar thing is with an abstract method:

```python
class AbstractFoo:

    def meth(self, _arg):
        raise NotImplementedError

class Foo(AbstractFoo):

    def meth(self, arg):
        do_stuff(arg)
```

**Comment by PCManticore** [comment]
In this case we might have a flag for not emitting when the name is prefixed by an underscore, being definitely known as not being used as a keyword argument. Would that be okay for you? Probably it would be off by default, I think.

**Comment by The-Compiler** [comment]
I think it would just need to be a flag for not emitting it when the name matches `dummy-variables-rgx`. 

Alternatively, I guess we'd need some alternative way to say "this argument is unused", other than the name. What about something like `# pylint: unused-arg=arg`?

**Comment by skortchmark9** [comment]
In this example, I think the `pylint-disable` comment should appear on the abstract class's method so you don't need to disable it for every child.
```
class AbstractFoo:

    def meth(self, _arg):
        raise NotImplementedError

class Foo(AbstractFoo):

    def meth(self, arg):
        do_stuff(arg)
```

**Comment by PCManticore** [comment]
Thank you @The-Compiler.
